packages/flutter_tools/lib/src/base/command_help.dart

```diff
@@ -0,0 +1,71 @@
+import 'dart:math' show max;
+
+import '../globals.dart';
+import 'io.dart' as io;
+import 'terminal.dart';
+
+const String fire = '🔥';
+const int maxLineWidth = 84;
+
+/// Encapsulates the help text construction and printing
+class CommandHelp {
+
+  CommandHelp._(this.key, this.description, [this.inParenthesis = '']);
+
+  factory CommandHelp.L() => CommandHelp._('L', 'Dump layer tree to the console.', 'debugDumpLayerTree');
+  factory CommandHelp.P() => CommandHelp._('P', 'Toggle performance overlay.', 'WidgetsApp.showPerformanceOverlay');
+  factory CommandHelp.R() => CommandHelp._('R', 'Hot restart');
+  factory CommandHelp.S() => CommandHelp._('S', 'Dump accessibility tree in traversal order.', 'debugDumpSemantics');
+  factory CommandHelp.U() => CommandHelp._('U', 'Dump accessibility tree in inverse hit test order.', 'debugDumpSemantics');
+  factory CommandHelp.a() => CommandHelp._('a', 'Toggle timeline events for all widget build methods.', 'debugProfileWidgetBuilds');
+  factory CommandHelp.d() => CommandHelp._('d', 'Detach (terminate "flutter run" but leave application running).');
+  factory CommandHelp.h() => CommandHelp._('h', 'Repeat this help message.');
+  factory CommandHelp.i() => CommandHelp._('i', 'Toggle widget inspector.', 'WidgetsApp.showWidgetInspectorOverride');
+  factory CommandHelp.o() => CommandHelp._('o', 'Simulate different operating systems.', 'defaultTargetPlatform');
+  factory CommandHelp.p() => CommandHelp._('p', 'Toggle the display of construction lines.', 'debugPaintSizeEnabled');
+  factory CommandHelp.q() => CommandHelp._('q', 'Quit (terminate the application on the device).');
+  factory CommandHelp.r() => CommandHelp._('r', 'Hot reload. $fire$fire$fire');
+  factory CommandHelp.s() => CommandHelp._('s', 'Save a screenshot to flutter.png.');
+  factory CommandHelp.t() => CommandHelp._('t', 'Dump rendering tree to the console.', 'debugDumpRenderTree');
+  factory CommandHelp.w() => CommandHelp._('w', 'Dump widget hierarchy to the console.', 'debugDumpApp');
+  factory CommandHelp.z() => CommandHelp._('z', 'Toggle elevation checker.');
+
+  /// The key associated with this command
+  final String key;
+  /// A description of what this command does
+  final String description;
+  /// Text shown in parenthesis to give the context
+  final String inParenthesis;
+
+  bool get _hasTextInParenthesis => inParenthesis != null && inParenthesis.isNotEmpty;
+
+  @override
+  String toString() {
+    final StringBuffer message = StringBuffer();
+    message.writeAll(<String>[terminal.bolden(key), description], ' ');
+
+    if (_hasTextInParenthesis) {
+      bool wrap = false;
+      final int maxWidth = max(io.stdio.terminalColumns ?? 0,  maxLineWidth);
```

Comment:
there's probably some code somewhere in this package that already abstracts over the terminalColumns logic

Reply:
I tried to find it, but this was what I came across. I will search further though.

Reply:
Something to do with `OutputPreferences` I think.

Reply:
Found it! Thanks! 😄